internal/routing/handlers/user_handler.go

```diff
@@ -437,3 +443,308 @@ func generateToken() string {
+
+	// Get the subscribeeId from the subscriptionId
+	queryString := "SELECT subscribee_id FROM alpha_schema.subscriptions WHERE subscription_id = $1"
+	row := tx.QueryRow(transactionCtx, queryString, subscriptionId)
+	var subscribeeId string
+	if err := row.Scan(&subscribeeId); err != nil {
+		if errors.Is(err, pgx.ErrNoRows) {
+			utils.WriteAndLogError(w, schemas.SubscriptionNotFound, http.StatusNotFound, errors.New("subscription not found"))
+			return
+		}
+		utils.WriteAndLogError(w, schemas.DatabaseError, http.StatusInternalServerError, err)
+		return
+	}
+
+	// Unsubscribe the user
+	queryString = "DELETE FROM alpha_schema.subscriptions WHERE subscriber_id = $1 AND subscribee_id = $2"
+	if _, err := tx.Exec(transactionCtx, queryString, jwtUserId, subscribeeId); err != nil {
+		if errors.Is(err, pgx.ErrNoRows) {
+			utils.WriteAndLogError(w, schemas.SubscriptionNotFound, http.StatusNotFound, errors.New("subscription not found"))
+			return
+		}
+		utils.WriteAndLogError(w, schemas.DatabaseError, http.StatusInternalServerError, err)
+		return
+	}
+
+	// Commit the transaction
+	if err := utils.CommitTransaction(w, tx, transactionCtx, cancel); err != nil {
+		return
+	}
+
+	w.WriteHeader(http.StatusNoContent)
+}
+
+// SearchUsers returns a list of users that match the search query using query parameters using offset and limit.
+func (handler *UserHandler) SearchUsers(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithDeadline(r.Context(), time.Now().Add(10*time.Second))
+	defer func() {
+		if err := ctx.Err(); err != nil {
+			log.Debug("Context error: ", err)
+		}
+		cancel()
+		log.Debug("Context canceled")
+	}()
+
+	// Get the search query from the query parameters
+	searchQuery := r.URL.Query().Get("username")
+
+	// Get the offset from the query parameters
+	offsetString := r.URL.Query().Get("offset")
+	if offsetString == "" {
+		offsetString = "0"
+	}
+	offset, err := strconv.Atoi(offsetString)
+	if err != nil {
+		utils.WriteAndLogError(w, schemas.BadRequest, http.StatusBadRequest, errors.New("offset invalid"))
+		return
+	}
+
+	// Get the limit from the query parameters
+	limitString := r.URL.Query().Get("limit")
+	if limitString == "" {
+		limitString = "10"
+	}
+	limit, err := strconv.Atoi(limitString)
+	if err != nil {
+		utils.WriteAndLogError(w, schemas.BadRequest, http.StatusBadRequest, errors.New("limit invalid"))
+		return
+	}
+
+	// Get the length of all users that match the search query
+	queryString := "SELECT COUNT(*) FROM alpha_schema.users"
+	row := handler.DatabaseManager.GetPool().QueryRow(ctx, queryString)
+	var records int
+	if err := row.Scan(&records); err != nil {
+		utils.WriteAndLogError(w, schemas.DatabaseError, http.StatusInternalServerError, err)
+		return
+	}
+
+	// Check if the offset is valid
+	if offset > records {
+		utils.WriteAndLogError(w, schemas.BadRequest, http.StatusBadRequest, errors.New("offset greater than records"))
+		return
+	}
```

Comment:
Versteh den Check hier nicht ganz? Die `records` in Postman beziehen sich auf die Gesamtheit der Records, die den Filterkriterien entsprechen. Hier schreibst du einfach alle User rein, die wir haben

Reply:
Wollte nur verhindern dass jemand den Offset über die Records hinaus setzt und dann keine Ergebnisse mehr hat